packages/experimental/Button/SPEC.md

```diff
@@ -72,12 +76,18 @@ The `Button` component can be loading if it's waiting for another action to occu
 
 ### Slots
 
+The `Button` component has three slots, or parts. The slots behave as follows:
+
 - `root` - The outer container representing the `Button` itself that wraps everything passed via the `children` prop.
 - `icon` - If specified, renders an `icon` either before or after the `children` as specified by the `iconPosition` prop.
 - `loader` - If specified, renders a `loader` before `children` while the `loading` flag is set to `true` in place of an icon. This slot is mutually exclusive to the icon slot.
```

Comment:
These don't match? In `Button.tsx` I see this?
```
  slots: {
    root: View,
    icon: Icon,
    content: Text,
  },
  ```
  
Glad we're documenting they exist!   In general though, I still feel Slots are hard to enforce as a public API... since we change the names a lot.

Reply:
Thanks for catching, will change.

Reply:
My worry here is that we don't really have any way to enforce people to _not_ change the slots. We pretty consistently add/modify slots anytime we make a change to a component. An example: Adding a chevron to `SubmenuItem` needed new slots for a `leftStack` and `rightStack`, which I eventually renamed to `startStack` and `endStack`, which could easily have been refactored now since I haven't checked it recently. Really, any change to a component means you have to change _something_ in the slot props / definition, which would affect anyone using `customize`.

That monologue is irrelevant to the PR though. I just realized I never typed that out anywhere.  So I'll just tag @ejlayne to note the expressed thought, and leave it aside.